### ISSUE TITLE
[🐛] Bad semver comparison causes reporting of vulnerability in fixed version

### ISSUE THREAD

**Issue by Niksko** [snyk/snyk#1722]
- `node -v`: v14.5.0
- `npm -v`: 7.6.3
- `snyk -v`: 1.439.0 (standalone)
- OS: OSX
- Command run: `snyk container test quay.io/kiwigrid/k8s-sidecar:1.10.7 --debug`

### Expected behaviour
No vulnerabilities reported

### Actual behaviour
Snyk reports a vulnerability

```
Testing quay.io/kiwigrid/k8s-sidecar:1.10.7...

✗ Medium severity vulnerability found in musl/musl-utils
  Description: Out-of-bounds Write
  Info: https://snyk.io/vuln/SNYK-ALPINE313-MUSL-1067865
  Introduced through: musl/musl-utils@1.2.2-r0, libc-dev/libc-utils@0.7.2-r3, meta-common-packages@meta
  From: musl/musl-utils@1.2.2-r0
  From: libc-dev/libc-utils@0.7.2-r3 > musl/musl-utils@1.2.2-r0
  From: meta-common-packages@meta > musl/musl@1.2.2-r0
  Fixed in: 1.2.2_pre2-r0
```

It was theorised [here](https://github.com/alpinelinux/docker-alpine/issues/123#issuecomment-776706144) that this may be an issue with the semver parsing. The fixed version `1.2.2_pre2-r0` is a pre-release version, and so `1.2.2-r0` should contain the fix.

Trivy reports no vulnerabilities

```
$ docker run --rm -v $HOME/Library/Caches:/root/.cache/ aquasec/trivy quay.io/kiwigrid/k8s-sidecar:1.10.7
2021-03-15T01:26:38.715Z        INFO    Detecting Alpine vulnerabilities...
2021-03-15T01:26:38.724Z        INFO    Trivy skips scanning programming language libraries because no supported file was detected

quay.io/kiwigrid/k8s-sidecar:1.10.7 (alpine 3.13.2)
===================================================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

### Steps to reproduce
Run the command above

### Debug log
```
  snyk test {
  _: [ 'quay.io/kiwigrid/k8s-sidecar:1.10.7', [Circular] ],
  debug: true,
  docker: true,
  experimental: true,
  callGraphBuilderTimeout: '300'
} +0ms
  snyk Docker.save: targetImage: quay.io/kiwigrid/k8s-sidecar:1.10.7, destination: /var/folders/bv/y724h_q90kg02rblrpq71dlm0000gp/T/tmp-89831-1iQ2kD2PYH1w/6c058820-8e7b-4722-acbb-d29e4adeb29d/image.tar +0ms
  snyk Potential policy locations found: [] +0msiwigrid/k8s-sidecar:1.10.7
  snyk sending request to: https://snyk.io/api/v1/test-dependencies +0ms
  snyk request body size: 12926 +0ms
  snyk gzipped request body size: 2674 +0ms
  snyk not using proxy +2ms
  snyk Can't remove folder /var/folders/bv/y724h_q90kg02rblrpq71dlm0000gp/T/tmp-89831-1iQ2kD2PYH1w/6c058820-8e7b-4722-acbb-d29e4adeb29d, got error null +0ms

Testing quay.io/kiwigrid/k8s-sidecar:1.10.7...

✗ Medium severity vulnerability found in musl/musl-utils
  Description: Out-of-bounds Write
  Info: https://snyk.io/vuln/SNYK-ALPINE313-MUSL-1067865
  Introduced through: musl/musl-utils@1.2.2-r0, libc-dev/libc-utils@0.7.2-r3, meta-common-packages@meta
  From: musl/musl-utils@1.2.2-r0
  From: libc-dev/libc-utils@0.7.2-r3 > musl/musl-utils@1.2.2-r0
  From: meta-common-packages@meta > musl/musl@1.2.2-r0
  Fixed in: 1.2.2_pre2-r0



Organization:      <redacted>
Package manager:   apk
Project name:      docker-image|quay.io/kiwigrid/k8s-sidecar
Docker image:      quay.io/kiwigrid/k8s-sidecar:1.10.7
Platform:          linux/amd64
Licenses:          enabled

Tested 36 dependencies for known issues, found 1 issue.

Pro tip: use `--file` option to get base image remediation advice.
Example: $ snyk test --docker quay.io/kiwigrid/k8s-sidecar:1.10.7 --file=path/to/Dockerfile

To remove this message in the future, please run `snyk config set disableSuggestions=true`
  snyk analytics {
  args: [
    'quay.io/kiwigrid/k8s-sidecar:1.10.7',
    {
      debug: true,
      docker: true,
      experimental: true,
      callGraphBuilderTimeout: '300',
      org: undefined,
      showVulnPaths: 'some'
    }
  ],
  command: 'test',
  org: undefined,
  metadata: {
    local: true,
    policies: 0,
    policyLocations: [],
    payloadSize: 12926,
    gzippedPayloadSize: 2674,
    depGraph: true,
    isDocker: true,
    'vulns-pre-policy': 3,
    vulns: 3,
    actionableRemediation: false,
    'error-message': 'Vulnerabilities found',
    error: undefined,
    'error-code': 'VULNS',
    command: 'test'
  },
  version: '1.439.0',
  os: 'macOS Catalina',
  nodeVersion: 'v12.18.1',
  standalone: true,
  integrationName: 'HOMEBREW',
  integrationVersion: '',
  integrationEnvironment: '',
  integrationEnvironmentVersion: '',
  id: '4d00084412efff1325d12aedbdfb6eb2213510e4',
  ci: false,
  environment: {},
  durationMs: 4074,
  metrics: {
    network_time: { type: 'timer', values: [Array], total: 1158 },
    cpu_time: { type: 'synthetic', values: [Array], total: 2916 }
  }
} +0ms

  snyk Exit code: 1 +0ms
  snyk sending request to: https://snyk.io/api/v1/analytics/cli +1s
  snyk request body size: 884 +0ms
  snyk gzipped request body size: 508 +0ms
  snyk not using proxy +0ms
```

**Comment by stuart-warren** [comment]
👍 
I'm also seeing and have done for a long time an issue with `jq`
```
✗ Medium severity vulnerability found in musl/musl-utils
  Description: Out-of-bounds Write
  Info: https://snyk.io/vuln/SNYK-ALPINE313-MUSL-1067865
  Introduced through: musl/musl-utils@1.2.2-r0, libc-dev/libc-utils@0.7.2-r3, meta-common-packages@meta
  From: musl/musl-utils@1.2.2-r0
  From: libc-dev/libc-utils@0.7.2-r3 > musl/musl-utils@1.2.2-r0
  From: meta-common-packages@meta > musl/musl@1.2.2-r0
  Image layer: Introduced by your base image (internal.docker.nexus.ocado.tech/internal-open-source/alpine-3.13.2:1.1.8@sha256:9b2a487ec2cb32bba5534176c4d0ff31445b7a0f0480cfecca42825532641b2d)
  Fixed in: 1.2.2_pre2-r0

✗ High severity vulnerability found in jq/jq
  Description: Out-of-Bounds
  Info: https://snyk.io/vuln/SNYK-ALPINE313-JQ-1067448
  Introduced through: jq/jq@1.6-r1
  From: jq/jq@1.6-r1
  Image layer: Introduced by your base image (internal.docker.nexus.ocado.tech/internal-open-source/alpine-3.13.2:1.1.8@sha256:9b2a487ec2cb32bba5534176c4d0ff31445b7a0f0480cfecca42825532641b2d)
  Fixed in: 1.6_rc1-r0
```

**Comment by RotemS** [comment]
Hi @Niksko , @stuart-warren, thanks for flagging this!
We've released a fix for this issue last week.
There's no need to update for it to be available (although always recommended 😃)
